homeassistant/components/matter/binary_sensor.py

```diff
@@ -95,6 +95,17 @@ def _update_from_device(self) -> None:
         entity_class=MatterBinarySensor,
         required_attributes=(clusters.OccupancySensing.Attributes.Occupancy,),
     ),
+    MatterDiscoverySchema(
+        platform=Platform.BINARY_SENSOR,
+        entity_description=MatterBinarySensorEntityDescription(
+            key="WaterLeakSensor",
```

Comment:
This matches the device type:

```suggestion
            key="WaterLeakDetector",
```